### DISCUSSION TITLE
DNS leaks with WE VPN

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uBlock-issues#3396] [Q&A]
Before uBO 1.60 when using a webExtension VPN in Firefox there were DNS leaks by default but could be fixed by disabling cname resolution.
Now there are _always_ DNS leaks no matter the cname setting. What to do?

**Comment by stephenhawk8054** [comment]
Can you go to advanced settings and change `dnsResolveEnabled` to `false`?

**Reply by ghost** [reply]
Sure. Is the behaviour above intended or bug?

**Reply by stephenhawk8054** [reply]
There's a new filter option [`ipaddress=`](https://github.com/gorhill/uBlock/commit/c6dedd253f) which utilizes Firefox [`dns.resolve()`](https://github.com/gorhill/uBlock/commit/6acf97bf51), hence a new advanced settings [`dnsResolveEnabled`](https://github.com/gorhill/uBlock/commit/760b2ffce6), so I think it's intended if you use VPN extensions.

But I'm not 100% sure, cc @gorhill

**Reply by gorhill** [reply]
With 1.60.0 there there should be no dns resolution when using a webextension-based VPN. What is that VPN extension you are using?

---

Also, which tool did you use to detect DNS leaks?

**Reply by ghost** [reply]
The extension is not publicly available but for testing [VPN from SandVPN](https://addons.mozilla.org/firefox/addon/sandvpn/) or [VPN add-on from uVPN.me](https://addons.mozilla.org/firefox/addon/uvpn-unlimited-vpn/) are sufficient.

Just looking at the DNS client events of windows.

Also mind that unrelated to this issue `network.dns.native_https_query = true` is causing leaks, too (fixed in FF132).

**Reply by gorhill** [reply]
As per [NoScript's own handling of `proxyInfo`](https://github.com/hackademix/noscript/blob/11.4.29/src/bg/RequestGuard.js#L477), `proxyInfo.type` must also be tested to avoid using `dns.resolve()`.

**Reply by gorhill** [reply]
Fix is in <https://github.com/gorhill/uBlock/commit/4305bfbdb10f7cccd3fabbff11f0510dce794afb>.

I will publish a 1.60.2 revision as soon as I can. Once you have that revision, you shouldn't need to disable `dnsResolveEnabled` in advanced settings.

---

If you can confirm all works fine with <https://github.com/gorhill/uBlock/releases/tag/1.60.1b2>, that would be appreciated.

**Reply by ghost** [reply]
Yes, 1.60.1b2 seems to fix it. Thank you.
